feat: detect vulnerable GitHub Actions (#1021)
oracle/macaron: docs/source/pages/tutorials/detect_vulnerable_github_actions.rst

.. Copyright (c) 2025 - 2025, Oracle and/or its affiliates. All rights reserved.
.. Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.

.. _detect-vuln-gh-actions:

=======================================
How to detect vulnerable GitHub Actions
=======================================

> This tutorial explains how to use a check in Macaron that detects vulnerable third-party GitHub Actions. This check is important for preventing security issues in your CI/CD pipeline, especially in light of recent incidents, such as vulnerabilities discovered in popular GitHub Actions like `tj-actions/changed-files <https://www.cve.org/CVERecord?id=CVE-2025-30066>`_, and `reviewdog/action-setup <https://www.cve.org/CVERecord?id=CVE-2025-30154>`_.

Reviewer: ```suggestion
This tutorial explains how to use a check in Macaron to detect vulnerable third-party GitHub Actions. This check is important for preventing security issues in your CI/CD pipeline, especially in light of recent incidents, such as vulnerabilities discovered in popular GitHub Actions: `tj-actions/changed-files <https://www.cve.org/CVERecord?id=CVE-2025-30066>`_, and `reviewdog/action-setup <https://www.cve.org/CVERecord?id=CVE-2025-30154>`_.
```

Author: See commit [a684508](https://github.com/oracle/macaron/pull/1021/commits/a6845083b70a170af22416b1f8cce556e16ca5dc).

> We will guide you on how to enable and use this check to enhance the security of your development pipeline.

Reviewer: ```suggestion
We will demonstrate how to enable and use this check to enhance the security of your development pipeline.
```

Author: See commit [a684508](https://github.com/oracle/macaron/pull/1021/commits/a6845083b70a170af22416b1f8cce556e16ca5dc).
